lib/vdsm/storage/blockVolume.py

```diff
@@ -365,6 +365,10 @@ def calculate_volume_alloc_size(
                 chunk_size_mb = config.getint("irs",
                                               "volume_utilization_chunk_mb")
                 alloc_size = chunk_size_mb * MiB
+                if alloc_size > capacity:
+                    # If the chunk size is bigger than the virtual size,
+                    # allocate the extent size instead.
+                    alloc_size = cls.align_size
```

Comment:
This will allocate 128M if you try to create 512M volume. What we really want is to allocate one chunk, but limit the chunk size to the capacity:

    alloc_size = max(chunk_size_mb * MiB, capacity)

The comment is also not helpful, it describes the code instead of why the code exist. Since this is trivial change, I don't think a comment is needed.

We need to add new test verifying that we allocated the right size in this case. The best way to do such changes it to add first the test, reproducing the bug. The test must fail to prove that the test is correct. Then add the fix which will fix the failure, proving that the fix is correct.

If the fix causes other tests to fail, it is best to fix the other failing tests assuming minimal allocation of one chunk (even if capacity < chunk) before you add the fix.

Reply:
`What we really want is to allocate one chunk, but limit the chunk size to the capacity` -> that would be min, no?
```python
    alloc_size = min(chunk_size_mb * MiB, capacity)
```
And that would make the allocation to be smaller than an extent if the capacity is smaller (as is the case in BZ). I can max it later with the align size, maybe?

And in this case, we already had a test covering the usecase in `blockvolume_test`: `TestBlockVolumeSize`.

Reply:
Yes I sometimes get min/max wrong, and this should be min().

Getting alloc_size < align_size is fine! lvm rounds the value up to the extent size
and create volume code adjust the size properly before writing the metadata.

You can add max(align_size, alloc_size) to align the value before calling lvm if you
think it makes the code more clear, but this may be needed in other cases, so best
do this later in all cases and not only for the case fixed here.

Reply:
```
Getting alloc_size < align_size is fine! lvm rounds the value up to the extent size
and create volume code adjust the size properly before writing the metadata.
```
That's good to know! Then I think the max is not really needed. Maybe a comment or a specific test would suffice. Thanks for the clarification.